pallets/payments/src/tests.rs

```diff
@@ -37,25 +29,35 @@ fn build_payment() -> Fees<Test> {
 		Some(remark.as_slice()),
 	);
 
-	assert_eq!(
-		PaymentStore::<Test>::get((SENDER_ACCOUNT, PAYMENT_BENEFICIARY, PAYMENT_ID)).unwrap(),
-		PaymentDetail {
+	if assertion == true {
```

Comment:
It's a bit weird to compare booleans to true/false when they can already be used for the check, instead I would suggest to name the variable in a more descriptive way like `if assert_payment_creation {}`.